frontend/src/pages/auth/Auth.tsx

```diff
@@ -8,43 +8,68 @@ import {
+import useForm from "./useForm";
 
-const Auth: React.FC = () => {
+const Auth = () => {
     const { eventType } = useParams();
+    const isSignUp = eventType === "signUp";
+    const { handleChange, values, handleSubmit, errors, touched } = useForm({ isSignUp });
 
     return (
         <SignupContainer>
-            <>
-                <Heading>
-                    {eventType === "signUp" ? "Rejestracja" : "Logowanie"}
-                </Heading>
-
-                <Form action="">
-                    <Label htmlFor="login">Login</Label>
-                    <Input type="text" id="login" name="login" />
-                    <Label htmlFor="password">Hasło</Label>
-                    <Input type="text" id="password" name="password" />
-                    {eventType === "signUp" && (
-                        <>
-                            <Label htmlFor="password">Powtórz hasło</Label>
-                            <Input type="text" id="password" name="password" />
-                        </>
-                    )}
-                </Form>
-
+            <Heading>{isSignUp ? "Rejestracja" : "Logowanie"}</Heading>
+            <Form onSubmit={handleSubmit}>
+                <Label htmlFor="login">Login</Label>
+                <Input
+                    type="text"
+                    id="login"
+                    name="login"
+                    onChange={handleChange}
+                    value={values.login}
+                />
+                {touched.login && errors.login && <p>{errors.login}</p>}
+                <Label htmlFor="password">Hasło</Label>
+                <Input
+                    type="password"
+                    id="password"
+                    name="password"
+                    onChange={handleChange}
+                    value={values.password}
+                />
+                {touched.password && errors.password && (
+                    <p>{errors.password}</p>
+                )}
+                {isSignUp && (
+                    <>
+                        <Label htmlFor="passwordConfirmation">
+                            Powtórz hasło
+                        </Label>
+                        <Input
+                            type="password"
+                            id="passwordConfirmation"
+                            name="passwordConfirmation"
+                            onChange={handleChange}
+                            value={values.passwordConfirmation}
+                        />
+                        {touched.passwordConfirmation &&
+                            errors.passwordConfirmation && (
+                                <p>{errors.passwordConfirmation}</p>
+                            )}
+                    </>
+                )}
                 <ButtonsContainer>
-                    <PrimaryButton>
-                        {eventType === "signUp" ? "Zarejestruj" : "Zaloguj"}
+                    <PrimaryButton type="submit">
```

Comment:
Button is default "submit", and there i see nothing override it, that's unnecassary.